celesta-sql/src/main/java/ru/curs/celesta/score/discovery/ScoreByScorePathDiscovery.java

```diff
@@ -49,15 +50,29 @@ public ScoreByScorePathDiscovery(String scorePath) {
 
     private Set<Resource> discoverScore(final File scoreDir) {
         try {
-            return Files.walk(scoreDir.toPath(), FileVisitOption.FOLLOW_LINKS)
+            Path scorePath = scoreDir.toPath().toAbsolutePath();
+            return Files.walk(scorePath, FileVisitOption.FOLLOW_LINKS)
                     .filter(path -> Files.isRegularFile(path) && path.toString().endsWith(".sql"))
-                    .map(Path::toAbsolutePath)
-                    .map(Path::toFile)
-                    .map(FileResource::new)
+                    .map(p -> {
+                        return new FileResource(p.toFile(), getNamespaceFromPath(scorePath.relativize(p)));
+                     })
                     .collect(Collectors.toCollection(LinkedHashSet::new));
         } catch (IOException ex) {
             throw new CelestaException(ex);
         }
     }
 
+    Namespace getNamespaceFromPath(Path p) {
```

Comment:
1. I generally dislike the idea of returning `null`. What does `null Namespace` mean? Maybe we could devise a special `DefaultNamespace` for this case? (However, I not insisting.)

2. I love `Collectors.joining`. How about
```java
 IntStream.range(0, p.getNameCount() - 1)
                .mapToObj(p::getName)
                .map(Path::toString)
                .map(String::toLowerCase)
                .collect(Collectors.joining("."));
```
?

Reply:
2. Done

Reply:
1. Introduced Namespace.DEFAULT